show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/components/MapViewModel.kt

```diff
@@ -0,0 +1,195 @@
+
+    // set up a graphic to indicate where the coordinates relate to, with an initial location
+    private val initialPoint by lazy {
+        Point(0.0, 0.0, SpatialReference.wgs84())
+    }
+    private val coordinateLocation by lazy {
+        Graphic(
+            geometry = initialPoint,
+            symbol = SimpleMarkerSymbol(
+                style = SimpleMarkerSymbolStyle.Cross,
+                color = Color.fromRgba(255, 255, 0, 255),
+                size = 20f
+            )
+        )
+    }
+
+    init {
+        // create a map that has the WGS 84 coordinate system and set this into the map
+        val basemapLayer = ArcGISTiledLayer(application.getString(R.string.basemap_url))
+        val map = ArcGISMap(Basemap(basemapLayer))
+        mapViewState.value.apply {
+            arcGISMap = map
+            graphicsOverlay.graphics.add(coordinateLocation)
+        }
+        // update the coordinate notations using the initial point
+        toCoordinateNotationFromPoint(initialPoint)
+    }
+
+    /**
+     * Updates the tapped graphic and coordinate notations using the [tappedPoint]
+     */
+    fun onMapTapped(tappedPoint: Point?) {
+        if (tappedPoint != null) {
+            // update the tapped location graphic
+            coordinateLocation.geometry = tappedPoint
+            mapViewState.value.graphicsOverlay.graphics.apply {
+                clear()
+                add(coordinateLocation)
+            }
+            // update the coordinate notations using the tapped point
+            toCoordinateNotationFromPoint(tappedPoint)
+        }
+    }
+
+    /**
+     * Uses CoordinateFormatter to update the UI with coordinate notation strings based on the
+     * given [newLocation] point to convert to coordinate notations
+     */
+    private fun toCoordinateNotationFromPoint(newLocation: Point) {
+        coordinateLocation.geometry = newLocation
+        // use CoordinateFormatter to convert to Latitude Longitude, formatted as Decimal Degrees
+        decimalDegrees.value = CoordinateFormatter.toLatitudeLongitudeOrNull(
+            point = newLocation,
+            format = LatitudeLongitudeFormat.DecimalDegrees,
+            decimalPlaces = 4
+        ) ?: return showErrorDialog()
+
+        // use CoordinateFormatter to convert to Latitude Longitude, formatted as Degrees, Minutes, Seconds
+        degreesMinutesSeconds.value = CoordinateFormatter.toLatitudeLongitudeOrNull(
+            point = newLocation,
+            format = LatitudeLongitudeFormat.DegreesMinutesSeconds,
+            decimalPlaces = 4
+        ) ?: return showErrorDialog()
+
+        // use CoordinateFormatter to convert to Universal Transverse Mercator, using latitudinal bands indicator
+        utm.value = CoordinateFormatter.toUtmOrNull(
+            point = newLocation,
+            utmConversionMode = UtmConversionMode.LatitudeBandIndicators,
+            addSpaces = true
+        ) ?: return showErrorDialog()
```

Comment:
This is a nitpick but currently these error dialogs will show "Invalid date format" but maybe it's better if the two types of errors have different messages like "Failure to convert to Utm Coordinates"